### PR TITLE
fix: threads now send correctly to discord.

### PR DESCRIPTION
**Pull request by Jaskowicz1**
This PR fixes an issue with threads, as they weren't sending correctly to discord [according to this part of the documentation](https://discord.com/developers/docs/resources/channel#:~:text=json%20params%20(thread)). This means threads can now be locked and archived correctly.

## Code change checklist

- [x] I have ensured that all methods and functions are **fully documented** using doxygen style comments.
- [x] My code follows the [coding style guide](https://dpp.dev/coding-standards.html).
- [x] I tested that my change works before raising the PR.
- [x] I have ensured that I did not break any existing API calls.
- [x] I have not built my pull request using AI, a static analysis tool or similar without any human oversight.

### CODE REVIEW

File: src/dpp/channel.cpp [hunk 1]
```diff
@@ -478,14 +478,34 @@ channel& channel::fill_from_json(json* j) {
 }
 
 std::string thread::build_json(bool with_id) const {
-	json j = json::parse(channel::build_json(with_id));
```
Comment (braindigitalis) [inline src/dpp/channel.cpp:481]: removing this line removes all attributes from channel we might want to keep like type, name, parent id, etc?

**Review by braindigitalis — CHANGES_REQUESTED** [review]

File: src/dpp/channel.cpp [hunk 2]
```diff
@@ -478,14 +478,34 @@ channel& channel::fill_from_json(json* j) {
...
-	j["type"] = (flags & CHANNEL_TYPE_MASK);
-	j["thread_metadata"] = this->metadata;
-	if (!this->applied_tags.empty()) {
-		j["applied_tags"] = json::array();
-		for (auto &tag_id: this->applied_tags) {
-			if (tag_id) {
-				j["applied_tags"].push_back(tag_id);
+	json j;
+	j["name"] = this->name;
```
Comment (braindigitalis) [inline src/dpp/channel.cpp:482]: name comes from the parent dpp::channel json

File: src/dpp/channel.cpp [hunk 3]
```diff
@@ -478,14 +478,34 @@ channel& channel::fill_from_json(json* j) {
...
+	j["archived"] = this->metadata.archived;
+	j["auto_archive_duration"] = this->metadata.auto_archive_duration;
+	j["locked"] = this->metadata.locked;
+	
+	if(is_private_thread()) {
+		j["invitable"] = this->metadata.invitable;
+	}
+
+	j["rate_limit_per_user"] = this->rate_limit_per_user;
```
Comment (braindigitalis) [inline src/dpp/channel.cpp:491]: rate_limit_per_user comes from the parent json

File: src/dpp/channel.cpp [hunk 4]
```diff
@@ -478,14 +478,34 @@ channel& channel::fill_from_json(json* j) {
...
+
+	if (is_forum() || is_media_channel()) {
```
Comment (braindigitalis) [inline src/dpp/channel.cpp:493]: is_forum or is_media_channel are set on the parent, are they not? not the thread within it?

File: src/dpp/channel.cpp (hunk 1 above)
Comment (Jaskowicz1) [reply src/dpp/channel.cpp:481]: Yeah, I completely forgot about the use case of creating threads. I'll see if we could maybe keep the old info whilst updating stuff.

File: src/dpp/channel.cpp (hunk 2 above)
Comment (Jaskowicz1) [reply src/dpp/channel.cpp:482]: 👍

Comment (Jaskowicz1) [reply src/dpp/channel.cpp:482]: 👍

File: src/dpp/channel.cpp (hunk 3 above)
Comment (Jaskowicz1) [reply src/dpp/channel.cpp:491]: 👍

**Review by braindigitalis — APPROVED** [review]

**Pull request merged**